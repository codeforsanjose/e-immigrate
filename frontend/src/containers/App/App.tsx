import React from 'react';
import MainContainer from '../MainContainer/MainContainer';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { Admin } from '../../compositions/Admin/Admin.js';
import { AdminDashboard } from '../AdminDashboard/AdminDashboard';
import { EditQuestionnaires } from '../EditQuestionnaires/EditQuestionnaires';
import './App.css';
// import { uploadQuestinnaires } from '../../sendRequest/apis';

export default function App() {
    return (
        <Router>
            <div className="App">
                <Routes>
                    <Route path="/dashboard">
                        <AdminDashboard 
                        />
                    </Route>
                    <Route path="/login">
                        <Admin />
                    </Route>
                    <Route path="/questionnaires">
                        <EditQuestionnaires />
                    </Route>
                    <Route path="/">
                        <MainContainer />
                    </Route>
                </Routes>
            </div>
        </Router>
    );
}
