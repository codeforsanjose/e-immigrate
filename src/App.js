import React, { useState, useEffect } from 'react';
import { languageOptions, contentVariations } from './LanguageContent';
import { ReactComponent as LanguagePicker } from './components/LanguagePicker.svg';
import HubspotForm from './components/HubspotForm';

function App() {

  const [language, setLanguage] = useState('en')
  
  
  const browserLanguage = window.navigator.userLanguage || window.navigator.language;
   
  useEffect(() => {
    if (browserLanguage){
      setLanguage(browserLanguage.substring(0,2))
    }
  },[browserLanguage])

  console.log('browserLanguage :>> ', browserLanguage);

  
  return (
    <div className='container d-flex flex-column align-items-center justify-content-center pt-4'>

      <h3>{ contentVariations[language].language }:</h3>
      <div className="btn-group">
        <button type="button" className="btn btn-md btn-primary dropdown-toggle d-flex align-items-center" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
          <LanguagePicker height='2rem' width='2rem' className='mr-2 '/>
          <span className='mr-2 d-flex-row align-items-center' style={ { verticalAlign: 'middle' }}>
            { contentVariations[language].value }
          </span>
        </button>
        <div className="dropdown-menu" aria-labelledby="dropdownMenuButton">
          { 
            languageOptions.map( (lang, idx) => {
              return (
                <button className='dropdown-item' value={lang.code} key={idx} onClick={ev => setLanguage(ev.target.value)} >{lang.full}</button>
              )
            })
          }
        </div>
      </div>

      <div className='my-5 mx-4 embed-responsive embed-responsive-16by9'>
        <iframe width="560" height="315" src={ contentVariations[language].video } title='Project New Citizen Video' frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen></iframe>
      </div>
      
      <HubspotForm hubspotFormId={ contentVariations[language].hubspotFormId } />
      
    </div>
  );
}

export default App;
