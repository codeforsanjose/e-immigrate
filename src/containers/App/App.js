import React from 'react';
import MainContainer from '../MainContainer/MainContainer';

import './App.css';

function App() {
  return (
    <div className='App'>
      <MainContainer />
    </div>
  );
}

export default App;
