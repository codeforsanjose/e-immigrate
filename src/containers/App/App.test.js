import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import renderer from 'react-test-renderer';

it('App renders correctly', () => {
    const tree = renderer
    .create(<App />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});