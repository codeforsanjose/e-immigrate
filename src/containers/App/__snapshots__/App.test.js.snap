// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="MainContainer"
  >
    <div
      className="wrapper"
    >
      <div
        className="items"
      >
        <div
          className="languageDropdown"
        >
          <div
            className="label"
          >
            <svg
              height="2rem"
              width="2rem"
            >
              LanguagePicker.svg
            </svg>
            <h4>
              Preferred Language
              :
            </h4>
          </div>
          <div>
            <select
              onChange={[Function]}
            >
              <option
                value="en"
              >
                English
              </option>
              <option
                value="es"
              >
                Español
              </option>
              <option
                value="vi"
              >
                Tiếng Việt
              </option>
            </select>
          </div>
        </div>
        <div>
          <div
            className="videoContainer"
          >
            <iframe
              allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
              allowFullScreen={true}
              className="video"
              frameBorder="0"
              height="315"
              src="https://www.youtube.com/embed/QOt3Jc59BD4"
              title="Project New Citizen Video"
              width="560"
            />
          </div>
        </div>
        <div
          id="hubspotForm"
        />
      </div>
    </div>
  </div>
</div>
`;
